config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java

```diff
@@ -67,6 +72,10 @@ protected final ApplicationContext getApplicationContext() {
 	 * @return the object that is chained after creating the {@link RequestMatcher}
 	 */
 	public C anyRequest() {
+		if (this.anyRequestConfigured) {
```

Comment:
Sorry, I didn't explain this well. `Assert.state` will do the if check for you as well, so we can remove that, too.

Reply:
> Sorry, I didn't explain this well. `Assert.state` will do the if check for you as well, so we can remove that, too.

Fixed.